### PR TITLE
[WIP] Use `RunningStage` in fetching.py

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Cleaning code after #9047

Fixes https://github.com/PyTorchLightning/pytorch-lightning/pull/9047#discussion_r694139133

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/utilities/fetching.py [hunk 1]
```diff
@@ -195,10 +192,11 @@ def __iter__(self) -> Generator[Tuple[Any, bool], None, None]:
         return self.fetching_function()
 
     def reset(self) -> None:
-        self.batches: List = []
-        self.dataloader: Optional[Iterable]
```
Comment (carmocca) [inline pytorch_lightning/utilities/fetching.py:199]: @tchaton the tests fail if this sets `self.dataloader = None`

What was your intention here?

If you meant to reset it, do we also need to reset `self.stage` and `self.dataloader_iter`?

Comment (tchaton) [reply pytorch_lightning/utilities/fetching.py:199]: Yes, ti shouldn't reset it. Let me remove it.
